File: benchmarks/method-call.rb
class Counter
    def reset()
        @number = 0
    end

    def up()
        @number += 1
    end
end

counter = Counter.new()
counter.reset()
1000000.times do
    counter.up()
    counter.up()
    counter.up()
    counter.up()
    counter.up()
    counter.up()
    counter.up()
    counter.up()
    counter.up()
    counter.up()
    counter.reset()
end

File: benchmarks/method-call.py
class Counter:
    def reset(self):
        self.number = 0
    def up(self):
        self.number += 1

def main():
    counter = Counter()
    counter.reset()
    for i in range(1000000):
        counter.up()
        counter.up()
        counter.up()
        counter.up()
        counter.up()
        counter.up()
        counter.up()
        counter.up()
        counter.up()
        counter.up()
        counter.reset()

main()